src/Components/Filtering/RangeFilter.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable max-len */
+import React, { useState } from 'react';
+import Button from '@mui/material/Button';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import propTypes from 'prop-types';
+import RangeFilterCard from './RangeFilterCard';
+import MultSelectElem from './Multiselect';
+
+const gradeRangeMetadata = ['0 to Pre-K', '1st', '2nd', '3rd', '4th', '5th', '6th', '7th', '8th', '9th', '10th', '11th', '12th'];
+const ageRangeMetadata = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18'];
+
+function RangeFilter({ setRangeState, setMultiSelectInput }) {
+  const [anchorEl, setAnchorEl] = useState(null);
+  // const [filterName, setFilterName] = useState('');
+  const [filterData, setFilterData] = useState({ age: { min: ageRangeMetadata[0], max: ageRangeMetadata[18] }, grade: { min: gradeRangeMetadata[0], max: gradeRangeMetadata[12] } });
+  const [tempMultiSelect, setTempMultiSelect] = useState({
+    Ethnicity: [],
+    Religion: [],
+    Gender: [],
+    Sexuality: [],
+  });
+  const open = Boolean(anchorEl);
+
+  const handleSave = () => {
```

Comment:
Final version will probably filter any time the user hits submit, so this state separation between the higher and lower component might not be necessary! Works great for now tho!